apps/advisor/pkg/app/checks/datasourcecheck/check.go

```diff
@@ -0,0 +1,100 @@
+package datasourcecheck
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	advisor "github.com/grafana/grafana/apps/advisor/pkg/apis/advisor/v0alpha1"
+	"github.com/grafana/grafana/apps/advisor/pkg/app/checks"
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/registry/apis/datasource"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/util"
+	"k8s.io/klog/v2"
+)
+
+func New(
+	datasourceSvc datasources.DataSourceService,
+	pluginStore pluginstore.Store,
+	pluginContextProvider datasource.PluginContextWrapper,
+	pluginClient plugins.Client,
+) checks.Check {
+	return &check{
+		DatasourceSvc:         datasourceSvc,
+		PluginStore:           pluginStore,
+		PluginContextProvider: pluginContextProvider,
+		PluginClient:          pluginClient,
+	}
+}
+
+type check struct {
+	DatasourceSvc         datasources.DataSourceService
+	PluginStore           pluginstore.Store
+	PluginContextProvider datasource.PluginContextWrapper
+	PluginClient          plugins.Client
+}
+
+func (c *check) Type() string {
+	return "datasource"
+}
+
+func (c *check) Run(ctx context.Context, obj *advisor.CheckSpec) (*advisor.CheckV0alpha1StatusReport, error) {
+	// Optionally read the check input encoded in the object
+	// fmt.Println(obj.Data)
+
+	dss, err := c.DatasourceSvc.GetAllDataSources(ctx, &datasources.GetAllDataSourcesQuery{})
+	if err != nil {
+		return nil, err
+	}
+
+	dsErrs := []advisor.CheckV0alpha1StatusReportErrors{}
+	for _, ds := range dss {
+		// Data source UID validation
+		err := util.ValidateUID(ds.UID)
+		if err != nil {
+			dsErrs = append(dsErrs, advisor.CheckV0alpha1StatusReportErrors{
+				Severity: advisor.CheckStatusSeverityLow,
+				Reason:   fmt.Sprintf("Invalid UID: %s", ds.UID),
+				Action:   "Change UID",
+			})
+		}
+
+		// Health check execution
+		id := &user.SignedInUser{OrgID: int64(1), Login: "admin"}
+		ctx = identity.WithRequester(ctx, id)
```

Comment:
Just curious, is this the way ™️  to make these kind of "background system calls" that require a user, or is this just to get things moving?

Reply:
Good point. It's probably not the best way but since we are [forcing admin-users only](https://github.com/grafana/grafana/pull/99440) it should be fine here. In any case, let me check if there is a good solution to do a better user impersonation for the request here.

Reply:
done, I'm using the `createdBy` annotation that gets attached to the resource, which seems to be enough!